Source/Core/DolphinQt2/MenuBar.cpp

```diff
@@ -716,3 +797,177 @@ void MenuBar::OnReadOnlyModeChanged(bool read_only)
 {
   m_recording_read_only->setChecked(read_only);
 }
+
+void MenuBar::ChangeDebugFont()
+{
+  bool okay;
+  QFont font = QFontDialog::getFont(&okay, Settings::Instance().GetDebugFont(), this,
+                                    tr("Pick a debug font"));
+
+  if (okay)
+    Settings::Instance().SetDebugFont(font);
+}
+
+void MenuBar::ClearSymbols()
+{
+  auto result = QMessageBox::warning(this, tr("Confirmation"),
+                                     tr("Do you want to clear the list of symbol names?"),
+                                     QMessageBox::Yes | QMessageBox::Cancel);
+
+  if (result == QMessageBox::Cancel)
+    return;
+
+  g_symbolDB.Clear();
+  Host_NotifyMapLoaded();
+}
+
+void MenuBar::GenerateSymbolsFromAddress()
+{
+  PPCAnalyst::FindFunctions(0x80000000, 0x81800000, &g_symbolDB);
+  Host_NotifyMapLoaded();
+}
+
+void MenuBar::GenerateSymbolsFromSignatureDB()
+{
+  PPCAnalyst::FindFunctions(0x80000000, 0x81800000, &g_symbolDB);
+  SignatureDB db(SignatureDB::HandlerType::DSY);
+  if (db.Load(File::GetSysDirectory() + TOTALDB))
+  {
+    db.Apply(&g_symbolDB);
+    QMessageBox::information(
+        this, tr("Information"),
+        tr("Generated symbol names from '%1'").arg(QString::fromStdString(TOTALDB)));
+    db.List();
+  }
+  else
+  {
+    QMessageBox::critical(
+        this, tr("Error"),
+        tr("'%1' not found, no symbol names generated").arg(QString::fromStdString(TOTALDB)));
+  }
+
+  Host_NotifyMapLoaded();
+}
+
+void MenuBar::GenerateSymbolsFromRSO()
+{
+  QString text = QInputDialog::getText(this, tr("Input"), tr("Enter the RSO module address:"));
+  bool good;
+  unsigned long address = text.toUInt(&good, 16);
+
+  if (good && address <= std::numeric_limits<u32>::max())
```

Comment:
Because of the above, this can just be inverted to 

```cpp
if (!good)
{
  QMessageBox::warning(this, tr("Error"), tr("Invalid RSO module address: %1").arg(text));
  return;
}
```